src/standalone/Vegetation/photosynthesis.jl

```diff
@@ -228,71 +237,170 @@ function update_photosynthesis!(p, Y, model::FarquharModel, canopy)
         s6,
         fC4,
     )
-    # TO DO: refactor to pass parameter struct, not the parameters individually
-    @. An = photosynthesis_at_a_point_Farquhar(
-        T_canopy,
-        β,
-        Rd,
-        f_abs * par_d / energy_per_mole_photon_par, # This function requires flux in moles of photons, not J
-        c_co2_air,
-        medlyn_factor,
-        R,
-        Vcmax25,
-        is_c3,
-        Γstar25,
-        ΔHJmax,
-        ΔHVcmax,
-        ΔHΓstar,
-        fC3,
-        fC4,
-        ΔHRd,
-        To,
-        θj,
-        ϕ,
-        oi,
-        Kc25,
-        Ko25,
-        ΔHkc,
-        ΔHko,
-        Q10,
-        s1,
-        s2,
-        s3,
-        s4,
-        s5,
-        s6,
-        E,
+    APAR_leaf_moles = @. lazy(
+        compute_APAR_leaf_moles(
+            f_abs,
+            par_d,
+            λ_γ_PAR,
+            lightspeed,
+            planck_h,
+            N_a,
+            LAI,
+        ),
+    )
+    A_leaf = @. lazy(
+        gross_leaf_photosynthesis_at_a_point_Farquhar(
+            T_canopy,
+            APAR_leaf_moles,
+            c_co2_air,
+            medlyn_factor,
+            R,
+            Vcmax25,
+            is_c3,
+            Γstar25,
+            ΔHJmax,
+            ΔHVcmax,
+            ΔHΓstar,
+            fC3,
+            fC4,
+            ΔHRd,
+            To,
+            θj,
+            ϕ,
+            oi,
+            Kc25,
+            Ko25,
+            ΔHkc,
+            ΔHko,
+            Q10,
+            s1,
+            s2,
+            s3,
+            s4,
+            s5,
+            s6,
+            E,
+        ),
+    )
+    # Compute net from gross
+    # See Table 11.5 of G. Bonan's textbook, Climate Change and Terrestrial Ecosystem Modeling (2019).
+    @. An = max(0, β * A_leaf - Rd)
```

Comment:
Farquhar saves this since conductance needs it, but pmodel doesnt need it ever, so it's not part of that cache